source/simulator/assembly.cc

```diff
@@ -1616,6 +1682,37 @@ namespace aspect
     FilteredIterator<typename DoFHandler<dim>::active_cell_iterator>
     CellFilter;
 
+    // determine which updates flags we need on cells and faces
+    const UpdateFlags cell_update_flags
+      = (update_values    |
+         update_gradients |
+         update_quadrature_points  |
+         update_JxW_values)
+        |
+        assemblers.stokes_system_assembler_properties.needed_update_flags;
+    const UpdateFlags face_update_flags
+      = (
+          // see if we need to assemble traction boundary conditions.
+          // only if so do we actually need to have an FEFaceValues object
+          parameters.prescribed_traction_boundary_indicators.size() > 0
```

Comment:
But we have the `assemblers.stokes_system_assembler_on_boundary_face_properties.needed_update_flags`. I can set that in `set_assemblers ()` and it seems to work. Or am I missing something?